zerver/lib/email_mirror.py

```diff
@@ -232,12 +233,14 @@ def extract_and_upload_attachments(message, realm):
         content_type = part.get_content_type()
         filename = part.get_filename()
         if filename:
-            s3_url = upload_message_image(filename, content_type,
-                                          part.get_payload(decode=True),
-                                          user_profile,
-                                          target_realm=realm)
-            formatted_link = u"[%s](%s)" % (filename, s3_url)
-            attachment_links.append(formatted_link)
+            attachment = part.get_payload(decode=True)
+            if isinstance(attachment, binary_type):
+                s3_url = upload_message_image(filename, content_type,
+                                              attachment,
+                                              user_profile,
+                                              target_realm=realm)
+                formatted_link = u"[%s](%s)" % (filename, s3_url)
+                attachment_links.append(formatted_link)
```

Comment:
Cool.  In general when writing logging, you should provide more details about what bad input was that one would want to debug or contact the user who sent the message (e.g. at least log the filename and sender of the email!).  I went ahead and fixed this to merge this commit.